File: client/web/js/lib/SHA256Salty.js
class SHA256Salty extends IHash {
    #saltSentence;
        
    constructor(saltSentence = "My own secret sentence for salt") {
        super();
        this.#saltSentence = saltSentence;
    }

    string = function(str /* :string */) {
        let hashArray = this.array(str);
        let res = ""
        for(let i = 0; i< hashArray.length; ++i)
        {
            if(hashArray[i] < 16)
                res += "0";
                res += hashArray[i].toString(16);
        }
        return res;
    }
    
    array = function(str /* :string */) {
        return this.postSalt(sha256.array(this.#preSalt(str)));
    }

    #preSalt = function(str /* :string */) {
        let res = "";
        for(let i = 0; i < str.length; ++i)
        {
            let saltChar = this.#saltSentence[i % this.#saltSentence.length];
            res += String.fromCharCode((str[i].charCodeAt() + i * saltChar.charCodeAt() ) % 256) 
            res += saltChar;
        }
        return res;
    }
    
    #postSalt = function(hashArray /* :Array<0..255> */) {
        //TODO: salt algorithm
        // I'm not sure, if I make a post salt algorithm, it will decrease the
        // SHA crash or not, so I don't do that
        return hashArray;
    }
} // end of class SHA256Salty extends HashInterface

File: src/python/sha256Salty/sha256Salty.py
import hashlib

class SHA256Salty():
    __saltSentence:str

    def __init__(self, saltSentence = "My own secret sentence for salt"):
        self.__saltSentence = saltSentence
    
    def string(self, s:str) -> str:
        s = self.preSalt(s)
        hs = hashlib.sha256( s.encode('utf-8') ).hexdigest()
        return self.postSalt(hs)
    
    def array(self, s:str):
        s = self.preSalt(s)
        hs = hashlib.sha256( s.encode('utf-8') ).digest()
        return [b for b in self.postSalt(hs)]

    def preSalt(self, s: str):
        res = ""
        for i in range(len(s)):
            saltChar = "\0"
            if  type(self.__saltSentence) == str and len(self.__saltSentence) != 0:
                saltChar = self.__saltSentence[i % len(self.__saltSentence)]
            res += chr(( ord(s[i]) + i * ord(saltChar) ) % 256) 
            res += saltChar
        return res
    
    def postSalt(self, hashArray):
        #TODO: salt algorithm
        # I'm not sure, if I make a post salt algorithm, it will decrease the
        # SHA crash or not, so I don't do that
        return hashArray

if __name__ == "__main__":
    s = SHA256Salty("alma")
    print(s.array("paprika"))
    print(str(s.array("paprika")).replace(" ",""))
    print(s.string("paprika"))